File: KufarCodeApp/Domain/UseCases/GetHomeLayoutUseCase.swift
import Combine

struct GetHomeLayoutUseCase {
    let cityRepository: CityRepository
    let layoutRepository: LayoutRepository

    func execute(cityCode: String, screen: String = "home", variant: String? = nil) -> AnyPublisher<CityLayout, Error> {
        layoutRepository.getLayout(cityCode: cityCode, screen: screen, variant: variant)
    }
}

File: android/app/src/main/java/com/kufarcode/app/domain/usecase/GetHomeLayoutUseCase.kt
package com.kufarcode.app.domain.usecase

import com.kufarcode.app.domain.repository.LayoutRepository

class GetHomeLayoutUseCase(private val repository: LayoutRepository) {
    operator fun invoke(cityCode: String, screen: String = "home", variant: String? = null) =
        repository.getLayout(cityCode, screen, variant)
}
